File: idea-plugin/src/main/java/org/ifinalframework/jetbrains/plugins/aio/api/yapi/model/Result.java
package org.ifinalframework.jetbrains.plugins.aio.api.yapi.model;


import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import lombok.Setter;


/**
 * Result
 *
 * @author iimik
 * @since 0.0.1
 **/
@Getter
@Setter
public class Result<T> {
    @JsonProperty("errcode")
    private String errCode;
    @JsonProperty("errmsg")
    private String errMsg;
    private T data;

    public boolean isSuccess() {
        return "0".equals(errCode);
    }
}

File: idea-plugin/src/main/kotlin/org/ifinalframework/plugins/jetbrains/aio/api/yapi/model/Result.kt
package org.ifinalframework.plugins.jetbrains.aio.api.yapi.model

import com.fasterxml.jackson.annotation.JsonProperty


/**
 * Result
 *
 * @author iimik
 * @since 1.6.0
 **/
data class Result<T>(
    @JsonProperty("errcode")
    val errCode: String,
    @JsonProperty("errmsg")
    val errMsg: String?,
    val data: T
) {
    fun isSuccess(): Boolean = errCode == "0"
}
